### DISCUSSION TITLE
Assembly version confusion - please help!

### DISCUSSION THREAD

**Discussion by RecumbentGourmet** [emgucv/emgucv#710] [Q&A]
Hi - I am receiving the following errors when I attempt to execute my Emgu/OpenCv application.

> Could not load file or assembly 'System.Drawing.Common, Version=6.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Here is some background info about my project that (hopefully) helps you understand.

I am working in Windows, creating a Visual Basic application that targets WinForms and .Net Framework V4.7.2.

The application is a part of a (very large) Visual Studio Solution that contains many projects.  Only 1 project (named EmguOpenCvUtils) contains any references to the Emgu assemblies.  It is a function library that contains methods the other applications can all call when they need specific support from OpenCV.   It compiles into it's own assembly.  It contains references to both Emgu.cv.runtime.windows and Emgu.CV.UI.  I am using Emgu V4.5.4.4788.  I tried updating to Emgu V4.5.5.4823, but I found the behavior was identical.

Currently I have two other projects in the solution that reference the EmguOpenCvUtils assembly.  One is a test application I put together only to test the methods I expose in the EmgiOpenCvUtils assembly.  The other is one of the main production applications for my user-base.

Both of these application target .Net framework 4.7.2, and both call the identical method in the EmguOpenCvUtils assembly.  Both applications compile and build with no errors.  At run-time, the test application does not give any errors and works properly, but the production application fails (with the above mentioned exception) as soon as it attempts to make a call to a method in the EmguOpenCvUtils assembly.

Here is the coder that invokes the error.  The 'FindCenterOfAperture' function is located in the EmguOpenCvUtils assembly.  The catch clause is invoked, without the code ever entering the FindCenterOfAperture method

> Private Sub FindAperture(imageBm As Bitmap)
        If Me.ApertureFinderCheckBox.Checked = True Then
            '
            ' [CSRT-387]  Make the area limits used by the aperture finder config file items
            '
            Try
                Dim areaLimits As ClerioRange(Of Integer) = ConfigEntries.BadalEntries.ApertureAreaLimits
                Me.FoundApertureObj = FindCenterOfAperture(imageBm, areaLimits.Minimum, areaLimits.Maximum)
            Catch ex As Exception
                Me.FoundApertureObj = Nothing
            End Try
        Else
            Me.FoundApertureObj = Nothing
        End If
    End Sub

For the life of me, I cannot see any difference between the two application's use of the EmguOpenCvUtils assembly, and yet one fails with this exception and the other works properly.  Any ideas about how I can sort this out?

**Comment by emgucv** [comment]
Emgu.CV.UI use System.Drawing.Common v6.0
One possible reason is that your other project use a different version of System.Drawing.Common. you can run nuget package manager on the solution to check. I do remember nuget package manager have issues with managing VB project base on .net framework, not sure if it affect yours.
